### DISCUSSION TITLE
Heading

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#220] [Component suggestions]
<img width="480" alt="visuele weergave van het heading component in verschillende groottes" src="https://github.com/nl-design-system/backlog/assets/545207/38befa82-de71-40c9-99a6-900a298da280">

We willen graag meer leren over het **Heading** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1708359932709109)

---

## Naam
Heading

## Doel
Koptekst die in de koppenstructuur ingesteld kan worden op het juiste niveau.

## Gerelateerde componenten
[Heading 1](https://github.com/orgs/nl-design-system/discussions/317), [Heading 2](https://github.com/orgs/nl-design-system/discussions/318), [Heading 3](https://github.com/orgs/nl-design-system/discussions/319), [Heading 4](https://github.com/orgs/nl-design-system/discussions/320), [Heading 5](https://github.com/orgs/nl-design-system/discussions/321), [Heading 6](https://github.com/orgs/nl-design-system/discussions/322), [Heading Group](https://github.com/orgs/nl-design-system/discussions/323), [Rich Text Content](https://github.com/orgs/nl-design-system/discussions/289)

## Richtlijnen

Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Heading](https://nldesignsystem.nl/heading)

## Onderzoek
[How Users Read on the Web - Nielsen Norman Group](https://www.nngroup.com/articles/how-users-read-on-the-web/)
Uit onderzoek blijkt dat mensen veelal scannen in plaats van lezen. Headings kunnen hier bij helpen.

[Headings Are Pick-Up Lines - Nielsen Norman Group](https://www.nngroup.com/articles/headings-pickup-lines/)
Artikel over het schrijven van een goede Heading. Niet vanuit onderzoek, eerder tips.

[Passive Voice Is Redeemed For Web Headings - Nielsen Norman Group](https://www.nngroup.com/articles/passive-voice-is-redeemed-for-web/)
Artikel, op basis van onderzoek, over het schrijven van Headings (en andere teksten).

[Teskt en taalgebruik richtlijnen - NL Design System](https://www.nldesignsystem.nl/richtlijnen/tekst-en-taalgebruik/#content-ontwerp)
Richtlijnen rondom tekst en taalgebruik.

[Accessible heading structure - The A11Y Project](https://www.a11yproject.com/posts/how-to-accessible-heading-structure/)
Artikel over het gebruik van Headings en hoe ze onder andere bijdragen aan toegankelijkheid.

## Richtlijnen
https://nldesignsystem.nl/wcag/2.4.6/

## Notities
- Zie [Slack bericht](https://codefornl.slack.com/archives/C7KDQRXMZ/p1708359932709109) voor input documentatie. Aantal H1's en semantiek ten opzichte van styling.
- Zie [Github Issue bij WCAG Audit Discussion](https://github.com/WCAG-Audit-Discussions/NL-BE/issues/61) voor de vraag over het gebruik van headings in een meerstappenformulier.

**Comment by sanderdegoose** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/apache-wicket-heading--docs)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=415-12207&mode=design&t=OzmZBqbwFOqKnQtT-4)

---

### Screenshots / Varianten

**Figma**
We gebruiken de H1 (32px) altijd voor pagina titels, maar we kunnen ook, mocht dat nodig zijn, de H1 overschrijven met een grotere letter (bijv. een 48px).

![Scherm­afbeelding 2024-02-21 om 12 51 15](https://github.com/nl-design-system/backlog/assets/545207/e5744b95-d2ec-400e-99ec-ce3908a225e6)

**Storybook**

Heading 1
<img width="1032" alt="Scherm­afbeelding 2024-02-21 om 14 08 51" src="https://github.com/nl-design-system/backlog/assets/545207/0c76e7db-3881-4274-b69e-9e40b06c5cb8">

Heading 2
<img width="1039" alt="Scherm­afbeelding 2024-02-21 om 14 08 57" src="https://github.com/nl-design-system/backlog/assets/545207/117d40c4-75f3-45d9-b515-160913ba4673">

Heading 3
<img width="1026" alt="Scherm­afbeelding 2024-02-21 om 14 09 11" src="https://github.com/nl-design-system/backlog/assets/545207/2e50b2a8-8870-4187-864e-f9514ead51b3">

Heading 4
<img width="1032" alt="Scherm­afbeelding 2024-02-21 om 14 09 17" src="https://github.com/nl-design-system/backlog/assets/545207/4bfe7977-2645-44b7-84cc-c815be2ad280">

Heading 5
<img width="1028" alt="Scherm­afbeelding 2024-02-21 om 14 09 22" src="https://github.com/nl-design-system/backlog/assets/545207/4bbb0c05-e154-48e5-942a-11308d49f142">

Heading 6
<img width="1035" alt="Scherm­afbeelding 2024-02-21 om 14 09 26" src="https://github.com/nl-design-system/backlog/assets/545207/c8e91687-3b45-4a89-bff8-a5d1f34a6190">


---

### Notities
- In Figma 9 sizes beschikbaar. 5 levels beschreven.
- In Figma opties voor Bold en Light font-weight inclusief richtlijnen over gebruik.
- In Storybook 6 sizes en levels.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook - Heading](https://designsystem.amsterdam/?path=/docs/components-text-heading--docs)
- [Storybook - Page Heading](https://designsystem.amsterdam/?path=/docs/components-text-page-heading--docs)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9734-12643&mode=design&t=Sn0TiS1YZL7izDxg-4)
- [GitHub](https://github.com/Amsterdam/design-system/blob/develop/packages/css/src/components/heading/heading.scss)

---

### Screenshots / Varianten
**Storybook**

Page Heading
![Scherm­afbeelding 2024-02-22 om 15 46 29](https://github.com/nl-design-system/backlog/assets/545207/c6d9d722-999d-42df-8cac-75c1b337ab9c)

Heading 1
![Scherm­afbeelding 2024-02-20 om 09 55 49](https://github.com/nl-design-system/backlog/assets/545207/677e8fcc-1b68-46db-9908-64b56fec7eef)

Heading 2
![Scherm­afbeelding 2024-02-20 om 09 55 53](https://github.com/nl-design-system/backlog/assets/545207/b3ba4bec-a740-4308-88e7-43d1eaad6658)

Heading 3
![Scherm­afbeelding 2024-02-20 om 09 55 58](https://github.com/nl-design-system/backlog/assets/545207/85c5a7fb-72bd-48ff-a1e5-a674a12c9ec5)

Heading 4
![Scherm­afbeelding 2024-02-20 om 09 56 02](https://github.com/nl-design-system/backlog/assets/545207/688e1e68-a05b-4115-a12e-638a1216f805)

Inverse colour
![Scherm­afbeelding 2024-02-20 om 09 56 07](https://github.com/nl-design-system/backlog/assets/545207/06e9d1ef-f3ee-48de-a358-048f070459f6)


**Figma**
Spacious
![Scherm­afbeelding 2024-02-20 om 10 08 14](https://github.com/nl-design-system/backlog/assets/545207/15528fa6-62f1-489c-8ecf-7b9a41c4f0ee)

Compact
![Scherm­afbeelding 2024-02-20 om 10 08 24](https://github.com/nl-design-system/backlog/assets/545207/bbf55c02-b4d6-463a-832f-19f1d6910335)

---

### Notities
- Responsive gedrag van fontsize obv clamp en calc in CSS. Hierover heeft Vincent ook verteld tijdens een [Heartbeat](https://youtu.be/Asd3RhL0nQY?si=PT6ESzAwv_0PK7dP).
- Storybook component biedt opties om 'level' (1 tot 4) en 'size' (1 tot 6) apart van elkaar in te stellen.
- Storybook component biedt de mogelijkheid om de kleur om te keren bij een donkere achtergrond.
- Apart Storybook component voor Page Heading. Dit is een variant van een Heading level 1 met een eigen stijl.
- Figma component biedt de mogelijkheid voor spacious en compact weergave.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-data-display-typography-heading-1--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Heading)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=3785-35&mode=design&t=dWSbPpZ8LEb9anPq-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=450-5078&mode=design&t=cfSA9SV8EbmmFc0h-4)

---

### Screenshots / Varianten

**Heading 1**
![Scherm­afbeelding 2024-02-20 om 10 25 20](https://github.com/nl-design-system/backlog/assets/545207/5d9574c1-403e-4e73-887e-ae9ed982a6eb)

**Heading 2**
![Scherm­afbeelding 2024-02-20 om 10 25 44](https://github.com/nl-design-system/backlog/assets/545207/1da683fe-de0b-4c71-a070-a535254bde90)

**Heading 3**
![Scherm­afbeelding 2024-02-20 om 10 25 52](https://github.com/nl-design-system/backlog/assets/545207/a83fdf3d-49fc-4db8-85cc-77819ec9b8bf)

**Heading 4**
![Scherm­afbeelding 2024-02-20 om 10 26 00](https://github.com/nl-design-system/backlog/assets/545207/49530994-35f0-496f-8122-ae21bfcaa1bb)

**Heading 5**
![Scherm­afbeelding 2024-02-20 om 10 26 09](https://github.com/nl-design-system/backlog/assets/545207/43124720-6be8-487c-8c9c-889d80ff2c33)


---

### Notities
- Zowel in Figma (Huidig) als Storybook combinaties van Heading met Paragraph.
- 5 sizes en levels beschreven.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/richtlijnen/typography)

---

### Screenshots / Varianten

![Scherm­afbeelding 2024-02-20 om 10 37 23](https://github.com/nl-design-system/backlog/assets/545207/e4bfe7b2-49dd-4acd-9532-b07cba4596f2)


---

### Notities
- 5 sizes en levels beschikbaar.
- Ook documentatie over hoe van een tekst een heading te maken en hoe een heading voorbij level 6 te creëren.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

- [Documentatie website](https://componenten.lenteveld.nl/paginatype-standaard)
- [Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**
![Scherm­afbeelding 2024-02-20 om 14 51 41](https://github.com/nl-design-system/backlog/assets/545207/5ac6185d-2f91-4e19-bb59-0280bb3a4e08)

**Sketch**
![Scherm­afbeelding 2024-02-20 om 14 50 06](https://github.com/nl-design-system/backlog/assets/545207/8f2dcfae-4041-48f8-8b77-8c8f53c936cd)

---

### Notities
- 4 sizes en levels beschikbaar.
- Headings ook als link beschikbaar.
- Headings binnen Sketch lijken om het niveau een andere kleur te hebben.

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook - Demo content pagina](https://nl-design-system.github.io/rvo/?path=/story/demo-pagina-s-algemeen-content--default)
- [Storybook - Heading](https://nl-design-system.github.io/rvo/?path=/docs/componenten-heading--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/heading)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4296-19249&mode=design&t=8PeRCJRP1cf7MspB-4)

---

### Screenshots / Varianten

**Storybook**
Heading 1
![Scherm­afbeelding 2024-02-21 om 14 26 17](https://github.com/nl-design-system/backlog/assets/545207/6bd626a5-93ef-4134-8187-a5d23782de15)

Heading 2
![Scherm­afbeelding 2024-02-21 om 14 26 22](https://github.com/nl-design-system/backlog/assets/545207/de4a0b69-eee8-4cb7-98d3-f1dfd73a827f)

Heading 3
![Scherm­afbeelding 2024-02-21 om 14 26 28](https://github.com/nl-design-system/backlog/assets/545207/2ec72daf-9be6-445f-bc00-c6d8edb1adb5)

Heading 4
![Scherm­afbeelding 2024-02-21 om 14 26 34](https://github.com/nl-design-system/backlog/assets/545207/a6049c9f-e4f5-4789-8b42-1a869e623764)

Heading 5
![Scherm­afbeelding 2024-02-21 om 14 26 39](https://github.com/nl-design-system/backlog/assets/545207/8c88ab5d-2808-4e3c-aba7-2aee99699f97)

Heading 6
![Scherm­afbeelding 2024-02-21 om 14 26 46](https://github.com/nl-design-system/backlog/assets/545207/3ed57e9d-e1ef-43ae-a43e-95c6f833f6ce)


---

### Notities
- 6 sizes en levels beschikbaar.
- In Storybook mogelijkheid om Headings als link toe te passen.
- In Storybook mogelijkheid om een Icon (before en after) toe te passen.
- Documentatie over beschikbaar om de heading zowel dikgedrukte als normale tekst mee te geven.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Documentatie website](https://nl-design-system.github.io/utrecht/docs/componenten/typografie/heading)
- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-heading-1--docs)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=450-5078&mode=design&t=9NaZalmzDb2VmbGe-4)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/heading-1/css)

---

### Screenshots / Varianten

**Storybook**

Heading 1
![Scherm­afbeelding 2024-02-21 om 14 38 38](https://github.com/nl-design-system/backlog/assets/545207/988559c6-f84c-4d9d-b435-184eba8d6dfa)

Heading 2
![Scherm­afbeelding 2024-02-21 om 14 38 47](https://github.com/nl-design-system/backlog/assets/545207/43fafa5f-de8b-47cc-ae34-3ec1c7a97792)

Heading 3
![Scherm­afbeelding 2024-02-21 om 14 38 55](https://github.com/nl-design-system/backlog/assets/545207/f5a321ec-4903-4e65-b479-72440c1a00c5)

Heading 4
![Scherm­afbeelding 2024-02-21 om 14 39 02](https://github.com/nl-design-system/backlog/assets/545207/4366649f-9ecd-48b7-9ebf-6135579004e3)

Heading 5
![Scherm­afbeelding 2024-02-21 om 14 39 11](https://github.com/nl-design-system/backlog/assets/545207/52bfbf6a-83bd-421b-9aa1-05f024fb6ff0)

Heading 6
![Scherm­afbeelding 2024-02-21 om 14 39 19](https://github.com/nl-design-system/backlog/assets/545207/8698959c-9ac6-4eb5-a4a0-172b7446cfa3)

---

### Notities
- Documentatie website bevat richtlijnen.
- 6 sizes en levels beschikbaar.
- Verschil tussen Figma en Storybook (Font size en Font weight vanaf H3 en lager)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Heading op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=heading)

**Comment by rianrietveld** [comment]
Op verzoek van @Robbert heb ik een klein experimentje gedaan hoe headings en `aria-level` wordt voorgelezen bij VO, NVDA en JAWS. Uitslag komt nog steeds overeen met wat [Adrian Roselli](https://adrianroselli.com/2024/05/level-setting-heading-levels.html) zegt: JAWS cannot handle any level above 6.

Zie: https://codepen.io/rianrietveld/pen/oNKPXJj?editors=1000

Resultaat:
H7 en hoger worden door geen van de drie screen readers als heading herkend en dat is logisch, want geen HTML.
VO en NVDA lezen `aria-level` 7 en hoger goed voor. Het dondert niet wat het niveau is, het wordt voorgelezen.
JAWS leest alles wat hoger is dan 6 voor als heading level 2. 

Dus, conclusie: `aria-level` groter dan 6 kan nu nog niet worden toegepast. Helaas.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Heading op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=heading+%23114)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Heading op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=heading+114)
